MAINTAINING.md

```diff
@@ -49,6 +49,89 @@ Note `--distribution-type all`.
 
 ### New IDE version
 
+Each new major platform release brings not only a lot of new features
+but usually also a bunch of new incompatibilities (both source and binary ones).
+Since we want to support at least the latest stable major platform release and new [EAP](https://www.jetbrains.com/resources/eap/)s,
+we have to compile the plugin with different platform versions to avoid any runtime errors.
+
+To solve the problem with incompatibilities, we use conditional compilation based on Gradle [source sets](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html).
+See [CONTRIBUTING.md](CONTRIBUTING.md#source-code-for-different-platform-versions) for more details about concrete source structure.
+
+#### How to support new platform version
+
+The following explanation uses `old`, `current` and `new` platform terms that mean:
+* `old` - number of the oldest supported platform version that should be dropped while
```

Comment:
```suggestion
* `old` - number of the oldest supported platform version that should be dropped
```
Or probably there is some missing word after `while`

Reply:
Thanks